app/controllers/api/databases.php

```diff
@@ -2032,13 +2046,12 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
             }
         }
 
-        if ($collection->getAttribute('permission') === 'collection') {
-            /** @var Document[] $documents */
-            $documents = Authorization::skip(fn () => $dbForProject->find('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $allQueries));
-            $total = Authorization::skip(fn () => $dbForProject->count('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $filterQueries, APP_LIMIT_COUNT));
-        } else {
+        if ($documentSecurity && !$valid) {
```

Comment:
@abnegate, is this `$valid` check required here? If so, why and is it also needed in the cursorDocument section above?

cc @Meldiron

Reply:
Yep it is needed; if `$valid`, user has collection level permissions and we should skip the document level check.

It is needed in the cursorDocument cursor and I have missed it, thank you :grin: